src/Struct/FindologicProduct.php

```diff
@@ -1169,4 +1167,21 @@ protected function addThumbnailUrl(MediaThumbnailEntity $media): void
             $this->images[] = new Image($encodedUrl, Image::TYPE_THUMBNAIL);
         }
     }
+
+    /**
+     * Goes through all given thumbnails, and will only add one thumbnail width as thumbnail. This avoids
+     * the adding of thumbnails in all various sizes.
+     */
+    protected function addThumbnailImages(MediaThumbnailCollection $thumbnails): void
+    {
+        $imageIds = [];
+        foreach ($thumbnails as $thumbnailEntity) {
+            if (in_array($thumbnailEntity->getMediaId(), $imageIds)) {
+                continue;
+            }
+
+            $this->addThumbnailUrl($thumbnailEntity);
```

Comment:
How are those sorted? Is the first one always the right one?

Reply:
Thumbnails are sorted by width, therefore the first thumbnail is the thumbnail with the smallest width:

https://github.com/findologic/plugin-shopware-6/blob/e990ecdea5199c439ffa71db276c0e88cbd81e98/src/Struct/FindologicProduct.php#L473